model-server/src/main/kotlin/org/modelix/model/server/handlers/ModelReplicationServer.kt

```diff
@@ -239,17 +240,29 @@ class ModelReplicationServer(
 
     override suspend fun PipelineContext<Unit, ApplicationCall>.postRepositoryObjectsGetAll(repository: String) {
         checkPermission(ModelServerPermissionSchema.repository(repository).objects.read)
-        runWithRepository(repository) {
-            val keys = call.receiveStream().bufferedReader().use { reader ->
-                reader.lineSequence().toHashSet()
-            }
-            val objects = withContext(Dispatchers.IO) { modelClient.store.getAll(keys) }.checkValuesNotNull {
-                "Object not found: $it"
+        val channel = call.receiveChannel()
+        val keys = hashSetOf<String>()
+        while (true) {
+            val line = channel.readUTF8Line() ?: break
+            keys.add(line)
+        }
+
+        val objects = runWithRepository(repository) {
+            withContext(Dispatchers.IO) {
+                modelClient.store.getAll(keys)
             }
-            call.respondTextWriter(contentType = ImmutableObjectsStream.CONTENT_TYPE) {
-                ImmutableObjectsStream.encode(this, objects)
+        }
+
+        for (entry in objects) {
+            if (entry.value == null) {
+                throw ObjectValueNotFoundException(entry.key)
             }
         }
+        @Suppress("UNCHECKED_CAST")
+        objects as Map<String, String>
```

Comment:
Why isn't this type already declared when the variable is created in line 250?

Reply:
In line 250 the type is `Map<String, String?>`. Only in line 257 (`entry.value == null`) we check that the values are not null.

Reply:
Ah, right. That missing bit of smart contracts in Kotlin.